### PR TITLE
Adds the Artemis helmet module, that protects from eye blur.

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

The Mk. 1 Artemis Helmet Module has been added to default vendors. It nullifies eye blurring from damage, pain, or otherwise.

image

image

Additionally, there is a Mk. 2 module that may be purchased from Req for 40 points or included in the experimental armor modules package, with a non-changed 400 points cost. This Mk. 2 includes the functionality of binoculars with the Artemis module.

image

image

The Mk. 2 Module is a subtype of the binoculars module for simplification.

Added a signal and component for when eye_blur is planemastered onto mobs.

Added a new component, blur_protection.
## Why It's Good For The Game

This is a small, non-intrusive to the gameplay loop module that is designed primarily for those who are sensitive to blurring, or those who just don't like it. It takes up a valuable slot (The helmet module) without truly severely impacting choice for those who do not wish for blur as there are ways around the different modules (Binoculars, rangefinders?) that are replaced by it, especially with the Mk. 2 version being purchasable.

 As far as I can tell, blur in TGMC is less of a true debuff and more of a aesthetic thing to imply the pain your character is feeling, as even at maximum blur things are still fairly visible. Due to this I believe this PR will not negatively impact gameplay, but rather positively by adding a non-intrusive, non-harmful option for those who suffer from eye strain problems.

image

Tentative Lumi approval of this PR's concept.

image
image
## Changelog
:cl:
add: Added a new Artemis helmet module to the base vendors that protects from eye blurring.
add: Added a upgraded Mk. 2 Artemis module to Req, purchasable for 40 points or included in the experimental modules crate.
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
thank you very mcuh

I have poor eyesight irl and having my character ahve blurry vision making me squint alot to see whats going on causing eyestrain is pretty bleh so this pr is a blessing

**Comment by Starfirejordan** [comment]
Good PR

**Comment by QualityVan** [comment]
Does this include immunity to the organ damage effect?

**Comment by Guidesu** [comment]
speed merge

**Comment by carlarctg** [comment]
> You should speed merge yourself, NOW! On the side note, does the MK2 Artemis module also protect from deafness couse I'm not sure if I read what you meant correctly in the pr.

no

**Comment by carlarctg** [comment]
wasn't sure if i wanted to do the deafness module and if so how

**Comment by Eskjjlj** [comment]
That would be great if it had screen shake protection too.

File: code/datums/components/blur_protection.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/datum/component/blur_protection
+	var/mob/living/protected_mob
+
```
Comment (TiviPlus) [inline code/datums/components/blur_protection.dm:3]: this could easily be an element

File: code/datums/components/blur_protection.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+/datum/component/blur_protection/Initialize(tint, tint_state)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
```
Comment (TiviPlus) [inline code/datums/components/blur_protection.dm:7]: passing args but not using them?

File: code/modules/clothing/modular_armor/attachments/modules.dm [hunk 3]
```diff
@@ -590,3 +590,28 @@
 	SIGNAL_HANDLER
 	beacon_datum = null
 
+/obj/item/armor_module/module/artemis
+	name = "Mark 1 Artemis Helmet Module"
```
Comment (TiviPlus) [inline code/modules/clothing/modular_armor/attachments/modules.dm:594]: improper/proper, could also make this Mk2 to shorten the name

File: code/modules/clothing/modular_armor/attachments/modules.dm [hunk 4]
```diff
@@ -590,3 +590,28 @@
...
+	slot = ATTACHMENT_SLOT_HEAD_MODULE
+	variants_by_parent_type = list(/obj/item/clothing/head/modular/marine/m10x = "artemis_head_xn", /obj/item/clothing/head/modular/marine/m10x/heavy = "artemis_head")
+	flags_attach_features = ATTACH_REMOVABLE|ATTACH_APPLY_ON_MOB
+	prefered_slot = SLOT_HEAD
+
+/obj/item/armor_module/module/artemis/on_attach(obj/item/attaching_to, mob/user)
+	. = ..()
+	parent.AddComponent(/datum/component/blur_protection)
+
+/obj/item/armor_module/module/binoculars/artemis_mk2 // a little cheating with subtypes
```
Comment (TiviPlus) [inline code/modules/clothing/modular_armor/attachments/modules.dm:608]: put this next to binocs + no numbers in typepaths pls

**Review by TiviPlus — APPROVED** [review]
Not gonna comment on the idea itself

**Review by TiviPlus — CHANGES_REQUESTED** [review]
.

File: code/datums/components/blur_protection.dm (hunk 1 above)
Comment (carlarctg) [reply code/datums/components/blur_protection.dm:3]: i have no idea what the distinction between these is

File: code/datums/components/blur_protection.dm (hunk 2 above)
Comment (carlarctg) [reply code/datums/components/blur_protection.dm:7]: whoops

File: code/modules/clothing/modular_armor/attachments/modules.dm (hunk 3 above)
Comment (carlarctg) [reply code/modules/clothing/modular_armor/attachments/modules.dm:594]: its for consistency between module names

**Comment by carlarctg** [comment]
comments handled ready for rereview

**Comment by Autisem** [comment]
id consider blurry vision a debuff but trading that in for the helmet module is not unreasonble

File: code/datums/components/blur_protection.dm (hunk 1 above)
Comment (TiviPlus) [reply code/datums/components/blur_protection.dm:3]: /datum/element vs /datum/component

Datum element creates one instance for all uses
component creates an instance every time it's added

**Comment by Tailiik** [comment]
It doesn't fucking attach to any of the helmets

**Pull request merged**